benchmarks/src/main/java/io/druid/benchmark/query/SearchBenchmark.java

```diff
@@ -0,0 +1,301 @@
+      InputRow row = gen.nextRow();
+      if (j % 10000 == 0) {
+        System.out.println(j + " rows generated.");
+      }
+      incIndex.add(row);
+    }
+
+    File tmpFile = File.createTempFile("SearchBenchmark-" + System.currentTimeMillis(), ".TEMPDIR");
+    tmpFile.delete();
+    tmpFile.mkdirs();
+    System.out.println(tmpFile.getAbsolutePath() + " isFile: " + tmpFile.isFile() + " isDir:" + tmpFile.isDirectory());
+    tmpFile.deleteOnExit();
+
+    File indexFile = INDEX_MERGER_V9.persist(
+        incIndex,
+        tmpFile,
+        new IndexSpec()
+    );
+
+    qIndex = INDEX_IO.loadIndex(indexFile);
+    factory = new SearchQueryRunnerFactory(
+        new SearchQueryQueryToolChest(
+            new SearchQueryConfig(),
+            QueryBenchmarkUtil.NoopIntervalChunkingQueryRunnerDecorator()
+        ),
+        QueryBenchmarkUtil.NOOP_QUERYWATCHER
+    );
+  }
+
+  @Setup(Level.Iteration)
+  public void setup2() throws IOException
+  {
+
+  }
+
+  @TearDown(Level.Iteration)
+  public void teardown() throws IOException
+  {
+  }
+
+  private IncrementalIndex makeIncIndex()
+  {
+    return new OnheapIncrementalIndex(
+        0,
+        QueryGranularity.NONE,
+        QueryBenchmarkUtil.INGESTION_AGGS,
+        true,
+        false,
+        true,
+        MAX_ROWS
+    );
+  }
+
+  public static final QuerySegmentSpec INTERVAL_SPEC = new MultipleIntervalSegmentSpec(
+      Arrays.asList(QUERY_INTERVAL)
+  );
+
+  private static final SearchQuery QUERY =
+      Druids.newSearchQueryBuilder()
+            .dataSource("blah")
+            .granularity(QueryGranularity.ALL)
+            .intervals(INTERVAL_SPEC)
+            .query("123")
+            .build();
+
+
+  private static <T> List<T> runQuery(QueryRunnerFactory factory, QueryRunner runner, Query<T> query)
+  {
+    QueryToolChest toolChest = factory.getToolchest();
+    QueryRunner<T> theRunner = new FinalizeResultsQueryRunner<>(
+        toolChest.mergeResults(toolChest.preMergeQueryDecoration(runner)),
+        toolChest
+    );
+
+    Sequence<T> queryResult = theRunner.run(query, Maps.<String, Object>newHashMap());
+    return Sequences.toList(queryResult, Lists.<T>newArrayList());
+  }
+
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.MICROSECONDS)
+  public void queryIncrementalIndex(Blackhole blackhole) throws Exception
+  {
+    QueryRunner<SearchHit> runner = QueryBenchmarkUtil.makeQueryRunner(
+        factory,
+        "incIndex",
+        new IncrementalIndexSegment(incIndex, "incIndex")
+    );
+
+    List<Result<SearchResultValue>> results = SearchBenchmark.runQuery(factory, runner, QUERY);
+    List<SearchHit> hits = results.get(0).getValue().getValue();
+    for (SearchHit hit : hits) {
+      blackhole.consume(hit);
+    }
+  }
+
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.MICROSECONDS)
+  public void queryQueryableIndex(Blackhole blackhole) throws Exception
```

Comment:
hmm, I think ideally we should have benchmarks for a few levels of things. this will help us track down performance improvements and regressions more easily, as well as optimize specific things more easily.

1. query a single incremental index (with `queryRunnerFactory.createRunner(...).run`) and materialize the results
2. query a single queryable index and materialize the results
3. query a bunch of queryable indexes with `mergeRunners` + `mergeResults` applied (like a historical would do), with a variable number of threads in the `mergeRunners` thread pool

1 and 2 help us test single-segment performance. 3 helps us test whole-query performance, including the merge step.